include/swift/ABI/TaskStatus.h

```diff
@@ -211,6 +210,33 @@ class TaskGroupTaskStatusRecord : public TaskStatusRecord {
     cur->childFragment()->setNextChild(child);
   }
 
+  void detachChild(AsyncTask *child) {
```

Comment:
Feels pretty silly, we never removed the tasks from the group record.

This must only be done by the group and the task running the group because we don't take locks here -- _on purpose_ and by design, only the parent task is allowed to access and mutate these things.

Reply:
Things worked without the removal because the tasks were leaking and never being deallocated, so as the group existed it'd hit cancelAll() which iterates over this record to cancel tasks. This was fine then since the tasks existed still.

As we fixed the bug about the child tasks lifecycle, and being properly released when they _fulfill_ a waiting tasks's value by copying it over, and then are destroyed... if we did a cancelAll() after a task had completed (and been destroyed), we'd iterate here and call cancel on trash memory.

So this PR fixes both: the leak, and the proper cancellation record handling